node/bundler.ts

```diff
@@ -186,19 +183,42 @@ const getBasePath = (sourceDirectories: string[], inputBasePath?: string) => {
   return commonPathPrefix(sourceDirectories)
 }
 
-export const addGeneratorFieldIfMissing = (
-  declaration: Declaration,
-  functions: EdgeFunction[],
-  internalFunctionsPath?: string,
-) => {
-  const fullFuncPath = functions?.find((func) => func.name === declaration.function)?.path
+interface MergeWithDeclarationConfigOptions {
+  name: string
+  config: FunctionConfig
+  declarations: Declaration[]
+}
 
-  // If function path is in the internalFunctionsPath, we assume it is autogenerated.
-  const isInternal = Boolean(internalFunctionsPath && fullFuncPath && isPathInside(fullFuncPath, internalFunctionsPath))
+// Merged any old-style non-in-source configs from declaration.
+const mergeWithDeclarationConfig = ({ name, config, declarations }: MergeWithDeclarationConfigOptions) => {
+  const declaration = declarations?.find((decl) => decl.function === name)
 
-  const generatorFallback = isInternal ? 'internalFunc' : undefined
-  return { ...declaration, generator: declaration.generator || generatorFallback }
+  return {
+    ...config,
+    name: declaration?.name || config.name,
```

Comment:
This slightly confused me, until i figured that this is the displayName and the passed in name is the functionName. Could we make this more clear? maybe alias the input `name` as `functionName` or add a comment here that this is the display name

Reply:
so we could do 
```suggestion
    name,
```
 here?

Reply:
I changed the way I named it now